File: VS Code Projects/Java/Magic.java
import java.util.*;
public class Magic {
    public static void main(String[] args) {
        Scanner sc=new Scanner (System.in);
        System.out.println("Enter a number : ");
        int n=sc.nextInt();
        int t=n,sum=0,r=0;
        while(t>9)
        {
            while(t>0)
            {
                r=t%10;
                sum=sum+r;
                t=t/10;
            }
            t=sum;
            sum=0;
        }
        if(t==1)
        {
            System.out.println("Magic ");
        }
        else
        {
            System.out.println("Not Magic");
        }
    }
}

File: VS Code Projects/C/Magic.c
#include <stdio.h>
int main()
{
    int n, t, r, sum = 0;
    printf("Enter a Number :");
    scanf("%d", &n);
    t = n;
    while (t > 9)
    {
        while (t > 0)
        {
            r = t % 10;
            sum = sum + r;
            t = t / 10;
        }
        t = sum;
        sum = 0;
    }
    if (t == 1)
    {
        printf("Magic Number");
    }
    else
    {
        printf("Not a Magic Number");
    }
    return 0;
}